src/main/java/xyz/wagyourtail/jsmacros/client/api/library/impl/FPlayer.java

```diff
@@ -158,4 +166,115 @@ public void takeScreenshot(String folder, String file, MethodWrapper<TextHelper,
                 if (callback != null) callback.accept(new TextHelper(text));
             });
     }
+
+    /**
+     * Creates a new PlayerInput object.
+     *
+     * @return a new {@link xyz.wagyourtail.jsmacros.client.api.classes.PlayerInput PlayerInput}.
+     * @see xyz.wagyourtail.jsmacros.client.api.classes.PlayerInput
+     * @since 1.4.x
+     */
+    public PlayerInput createPlayerInput() {
+        return new PlayerInput();
+    }
+
+    /**
+     * Creates a new {@code PlayerInput} object with the current inputs of the player.
+     *
+     * @return a new {@link xyz.wagyourtail.jsmacros.client.api.classes.PlayerInput PlayerInput} with the current inputs.
+     * @see xyz.wagyourtail.jsmacros.client.api.classes.PlayerInput
+     * @since 1.4.x
+     */
+    public PlayerInput getCurrentPlayerInput() {
+        return new PlayerInput(mc.player.input, mc.player.yaw, mc.player.pitch, mc.player.isSprinting());
+    }
+
+    /**
+     * Adds a new {@code PlayerInput} to {@code MovementQueue} to be executed
+     *
+     * @param input the PlayerInput to be executed
+     * @see xyz.wagyourtail.jsmacros.client.movement.MovementQueue
+     * @since 1.4.x
+     */
+    public void addInput(PlayerInput input) {
+        MovementQueue.append(input, mc.player);
+    }
+
+    /**
+     * Adds multiple new {@code PlayerInput} to {@code MovementQueue} to be executed
+     *
+     * @param inputs the PlayerInputs to be executed
+     * @see xyz.wagyourtail.jsmacros.client.movement.MovementQueue
+     * @since 1.4.x
+     */
+    public void addInputs(List<PlayerInput> inputs) {
+        for (PlayerInput input : inputs) {
+            addInput(input);
+        }
+    }
+
+    /**
+     * Predicts where one tick with a {@code PlayerInput} as input would lead to.
+     *
+     * @param input the PlayerInput for the prediction
+     * @param draw  whether to visualize the result or not
+     * @return the position after the input
+     * @since 1.4.x
+     */
+    public Vec3d predictInput(PlayerInput input, boolean draw) {
+        return predictInputs(Collections.singletonList(input), draw).get(0);
+    }
+
+    /**
+     * Predicts where each {@code PlayerInput} executed in a row would lead
+     *
+     * @param inputs the PlayerInputs for each tick for the prediction
+     * @param draw   whether to visualize the result or not
+     * @return the position after each input
+     * @since 1.4.x
+     */
+    public List<Vec3d> predictInputs(List<PlayerInput> inputs, boolean draw) {
```

Comment:
add overloads `predictInputs(List<PlayerInput> inputs)` with draw `false` by default

Reply:
I can do that